include/dfu/flash_img.h

```diff
@@ -20,6 +20,32 @@ struct flash_img_context {
 	struct stream_flash_ctx stream;
 };
 
+/**
+ * @brief Enum which defines all verification algorithms
+ *
+ * Algorithm list:
+ * - SHA-256
+ */
+enum flash_img_check_alg {
+	FLASH_IMG_CHECK_ALG_SHA256,
+};
+
+/**
+ * @brief Structure for verify flash region integrity
+ *
+ * This is used to pass data and select algorithm to be used to check flash
+ * integrity.
+ *
+ * Note: match vector length is fixed and depends on size from hash algorithm
+ * used to verify flash integrity.
+ */
+struct flash_img_check {
+	enum flash_img_check_alg alg;	/** Match algorithm */
+	const uint8_t *match;		/** Match vector data */
+	size_t len;			/** Content len to be compared */
+	size_t off;			/** Content start point offset */
```

Comment:
I'd prefer this to be abstracted so `dfu` could handle it by itself. It would reduce the amount of information the API user needs to find prior to making use of this functionality. Essentially, if it is on bank 0 or 1 should be enough for `dfu` to figure out which offset to use, in a generic way.